### ISSUE TITLE
Parameters set in "Custom run" are sometimes ignored

### ISSUE THREAD

**Issue by przecze** [PrefectHQ/prefect#10217]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

If there's a param value override defined using UI "edit" option for deployment, the value provided by the user manually in "Custom Run" is ignored

### Reproduction

```python3
Repr steps
1. Register a deployment with default value of parameter `param1:` (string) set to A
2. In prefect UI go to deployment->edit and change the `param1: ` value to B
3. In prefect UI go to deployment->run->"Custom Run" and set the `param1:` value to A and start the run
Behavior:
Started run uses `B` as value for `param1`
Expected:
Started run uses `A` as value for `param1`
```

### Error

```python3
Behavior:
Started run uses `B` as value for `param1`
Expected:
Started run uses `A` as value for `param1`
```

### Versions

```Text
Version:             2.10.8
API version:         0.8.4
Python version:      3.10.11
Git commit:          79093235
Built:               Mon, May 8, 2023 12:23 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         server
```

**Comment by zhen0** [comment]
@przecze - Thanks for the issue! We've made some updates to the custom run form.  Could you try updating to the latest version and see if this is still an issue for you?

**Comment by przecze** [comment]
fixed after upgrading server and agent to 2.10.21. Thanks!
